File: java/03-bootstrap/src/main/java/com/fResult/bootstrap/common/utils/DataSourceUtils.java
package com.fResult.bootstrap.common.utils;

import javax.sql.DataSource;
import org.springframework.core.io.ClassPathResource;
import org.springframework.jdbc.datasource.init.DatabasePopulatorUtils;
import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;

public final class DataSourceUtils {
  private DataSourceUtils() {
    throw new IllegalStateException("This cannot be instantiated");
  }

  public static DataSource initializeDdl(final DataSource dataSource) {
    final var populator = new ResourceDatabasePopulator(new ClassPathResource("/schema.sql"));
    DatabasePopulatorUtils.execute(populator, dataSource);

    return dataSource;
  }
}

File: kotlin/03-bootstrap/src/main/kotlin/com/fResult/bootstrap/common/utils/DataSourceUtils.kt
package com.fResult.bootstrap.common.utils

import org.springframework.core.io.ClassPathResource
import org.springframework.jdbc.datasource.init.DatabasePopulatorUtils
import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator
import javax.sql.DataSource

class DataSourceUtils private constructor() {
  companion object {
    fun initializeDdl(dataSource: DataSource): DataSource {
      val populator = ResourceDatabasePopulator(ClassPathResource("/schema.sql"))
      DatabasePopulatorUtils.execute(populator, dataSource)

      return dataSource
    }
  }
}

